tests/manual/RichTextAreaDemo/src/com/oracle/demo/richtext/codearea/DemoSyntaxDecorator.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2023, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.demo.richtext.codearea;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javafx.scene.paint.Color;
+import jfx.incubator.scene.control.richtext.SyntaxDecorator;
+import jfx.incubator.scene.control.richtext.TextPos;
+import jfx.incubator.scene.control.richtext.model.CodeTextModel;
+import jfx.incubator.scene.control.richtext.model.RichParagraph;
+import jfx.incubator.scene.control.richtext.model.StyleAttributeMap;
+
+/**
+ * Simple {@code SyntaxDecorator} which emphasizes digits and java keywords.
+ * This is just a demo.
+ */
+public class DemoSyntaxDecorator implements SyntaxDecorator {
+    private static final StyleAttributeMap DIGITS = StyleAttributeMap.builder().setTextColor(Color.MAGENTA).build();
+    private static final StyleAttributeMap KEYWORDS = StyleAttributeMap.builder().setTextColor(Color.GREEN).build();
+    private static Pattern PATTERN = initPattern();
+
+    public DemoSyntaxDecorator() {
+    }
+
+    @Override
+    public String toString() {
+        return "DemoSyntaxDecorator";
+    }
+
+    @Override
+    public RichParagraph createRichParagraph(CodeTextModel model, int index) {
```

Comment:
When implementing "real" syntax highlighting, the style of a paragraph will depend on the style of it's predecessors (e.g. multiline strings or multiline comments)...

Reply:
Correct, this is just a demo.

The information passed to the **SyntaxDecorator** interface should be sufficient to support any "real" syntax, at least I hope it is.  We are passing the model to each method, so it can re/construct the syntax or cache it as needed.

Reply:
Consider this situation: When a user types some text into paragraph 2, it may change the styling in paragraph 2, 3, 4, ... 
Will the createRichParagraph - method be called for each paragraph after each typed character? That would probably be inefficient; On the other hand, if it is only called for the paragraph the user typed in, that may not be enough.

Reply:
In this case the model fires a change event which defines the affected area.  The view (the skin) determines which visible elements need to be updated, by calling StyledTextModel::getParagraph().

I don't think there will be efficiency issues because
1. these visuals are likely to be updated anyway
2. the number of requested paragraphs is limited by the size of the sliding window (i.e. screen size + some margin)